src/components/TokenField/TokenSelector.tsx

```diff
@@ -1,67 +1,30 @@
 import { Combobox } from "@headlessui/react";
-import { useState } from "react";
 import { SelectorProps } from "./types";
-import Image from "components/Image";
-import Icon, { DrawerIcon } from "../Icon";
+import { DrawerIcon } from "../Icon";
+import TokenOptions from "./TokenOptions";
 
 export default function TokenSelector({
+  selectedChainId,
   onChange,
-  tokens,
-  token,
+  selectedToken,
 }: SelectorProps) {
-  const [searchText, setSearchText] = useState("");
-
-  const filtered =
-    searchText === ""
-      ? tokens
-      : tokens.filter((t) => {
-          return t.symbol.includes(searchText.toLowerCase());
-        });
-
   return (
     <Combobox
-      value={token}
+      value={selectedToken}
       onChange={onChange}
       as="div"
       className="flex items-center gap-1 w-full dark:text-gray"
     >
-      <span className="text-sm">{token.symbol}</span>
+      <span className="text-sm">{selectedToken.symbol}</span>
 
-      {tokens.length > 1 && (
-        <Combobox.Button>
-          {({ open }) => <DrawerIcon isOpen={open} size={20} />}
-        </Combobox.Button>
-      )}
+      <Combobox.Button>
+        {({ open }) => <DrawerIcon isOpen={open} size={20} />}
+      </Combobox.Button>
 
-      <Combobox.Options className="absolute right-0 top-2 z-10 border border-prim p-1 mt-10 max-h-60 w-max overflow-y-auto rounded-md bg-gray-l5 dark:bg-blue-d7 shadow-lg focus:outline-none">
```

Comment:
TokenOptions as a standalone component, fetches tokens rather than tokens being passed into it